power_grid_model_c/power_grid_model/include/power_grid_model/component/transformer.hpp

```diff
@@ -53,6 +52,13 @@ class Transformer : public Branch {
               calculate_z_pu(transformer_input.r_grounding_from, transformer_input.x_grounding_from, u1_rated)},
           z_grounding_to_{
               calculate_z_pu(transformer_input.r_grounding_to, transformer_input.x_grounding_to, u2_rated)} {
+        // init tap_pos_ linter smell
+        if (transformer_input.tap_pos == na_IntS) {
+            tap_pos_ = transformer_input.tap_nom == na_IntS ? IntS{0} : transformer_input.tap_nom;
+        } else {
+            tap_pos_ = transformer_input.tap_pos;
+        }
+
```

Comment:
I think this can go in the transformer_utils.hpp since  its repeated

Reply:
As I tried to do this, i saw there is a whole lot of other duplicated code too. A good amount of refactoring would be needed here. Hence lets do that later

Reply:
Could use an immediate jira ticket ;)